src/main/java/core/basesyntax/supplier/FigureSupplier.java

```diff
@@ -0,0 +1,52 @@
+package core.basesyntax.supplier;
+
+import core.basesyntax.model.Circle;
+import core.basesyntax.model.Figure;
+import core.basesyntax.model.IsoscelesTrapezoid;
+import core.basesyntax.model.Rectangle;
+import core.basesyntax.model.RightTriangle;
+import core.basesyntax.model.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Circle circle = new Circle("WHITE", 10);
```

Comment:
In that case you will return the same object every time, let's return a new circle in getDefaultFigure()